1/storing-a-value.md

```diff
@@ -32,69 +32,74 @@ Here is an example of how you would store an `AccountId` and `Balance`:
 
-You can find all the supported Substrate types in [`core/env/types.rs`](https://github.com/paritytech/ink/blob/master/core/src/env/types.rs).
-
-## Initializing Storage
-
-> **IMPORTANT:** This section is important. Read it twice. Then have tea and read it again.
-
-Before you can interact with any storage items in an ink! contract, **you must make sure they are initialized!** If you do not initialize a storage item and try to access it, your contract call will not succeed and any state changes caused by the transaction will be reverted. (Gas fees will still be charged though!)
-
-For storage values like the ones above, we can set an initial value with:
-
-```rust
-self.my_bool.set(false);
-self.my_number.set(42);
-self.my_account.set(AccountId::from([0x1; 32]));
-self.my_balance.set(1337);
-```
-
-This can be done anywhere in our contract logic, but most commonly this happens by using the `#[ink(constructor)]` attribute.
+You can find all the supported Substrate types in [`crates/storage/src/lib.rs`](https://github.com/paritytech/ink/blob/master/crates/storage/src/lib.rs).
 
 ## Contract Deployment
 
-Every ink! smart contract must have a constructor which is run once when a contract is created. ink! smart contracts can have multiple different constructors:
+Every ink! smart contract must have a constructor which is run once when a contract is created. ink! smart contracts can have multiple constructors:
 
 ```rust
 use ink_lang as ink;
 
-#[ink::contract(version = "0.1.0")]
-impl MyContract {
+#[ink::contract()]
+mod my_contract {
     #[ink(storage)]
-    struct MyContract {
-        number: storage::Value<u32>,
+    pub struct MyContract {
+        number: u32,
     }
 
     impl MyContract {
         #[ink(constructor)]
-        fn new(&mut self, init_value: u32) {
-            self.number.set(init_value);
+        pub fn new(init_value: u32) -> Self {
+            Self {
+                number: init_value,
+            }
         }
 
         #[ink(constructor)]
-        fn default(&mut self) {
-            self.new(0)
+        pub fn default() -> Self {
+            Self {
+                number: Default::default(),
+            }
         }
     /* --snip-- */
     }
 }
 ```
 
+## Initializing Storage
+
+> **IMPORTANT:** This section is important. Read it twice. Then have tea and read it again.
+
+Before you can interact with any storage items in an ink! contract, **you must make sure they are initialized!** If you do not initialize a storage item and try to access it, your contract call will not succeed and any state changes caused by the transaction will be reverted. (Gas fees will still be charged though!)
+
+Notice that in the example above, the initial value of the item is set in the constructor:
+
+```rust
+pub fn new(init_value: u32) -> Self {
+    Self {
+        number: init_value,
+    }
+}
+```
+
+This can be done anywhere in our contract logic, but most commonly this happens by using the `#[ink(constructor)]` attribute.
+
```

Comment:
This tip applied where the signature of an ink! constructor was still `&mut self, ...` but nowadays this rule is guarded by ink! and the Rust compiler and users generally can no longer easily shoot their foot by forgetting to initialize a value since they must initialize everything always since they otherwise run into compile errors.

Reply:
Should this section be removed?

Reply:
yey